packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -220,15 +191,24 @@ if (!ExecutionEnvironment.canUseDOM) {
 
     let currentTime = now();
     // Next, as long as we have idle time, try calling more callbacks.
-    while (frameDeadline - currentTime > 0 && pendingCallbacks.length > 0) {
-      const latestCallbackConfig = pendingCallbacks.shift();
+    while (
+      frameDeadline - currentTime > 0 &&
+      headOfPendingCallbacksLinkedList !== null
+    ) {
+      const latestCallbackConfig = headOfPendingCallbacksLinkedList;
+      // move head of list to next callback
+      headOfPendingCallbacksLinkedList =
+        latestCallbackConfig.nextCallbackConfig;
+      if (headOfPendingCallbacksLinkedList) {
+        headOfPendingCallbacksLinkedList.previousCallbackConfig = null;
+      }
       frameDeadlineObject.didTimeout = false;
       const latestCallback = latestCallbackConfig.scheduledCallback;
-      const newCallbackId = latestCallbackConfig.callbackId;
-      safelyCallScheduledCallback(latestCallback, newCallbackId);
+      // TODO: error handling
```

Comment:
So it's ok to land this without error handling because React is still the only consumer, but we'll need to handle errors correctly before we can use this anywhere else. Is that right?